### PR TITLE
extract hyperpack onnx model (automl flavor)

### PR DESCRIPTION
**Pull request by godsylla**
## Summary

Given a hyperpack, 
- unzip the hyperpack onnx model saved therein
- for an automl model, convert the underlying onnx model to a pytorch model 

_Describe your changes._
- added `UnpackHyperpack` class to `utilities.py` in `hyperpackage` directory
- added `automl` to `hyperpackage/flavor/` directory for code to unpack automl models 

## Testing
- checkout this branch
- install hyperpackage with the following pip installation instruction:
    - `%pip install git+https://github.com/gohypergiant/hyperdrive.git@MLSDK472-extract-hp-onnxmodel#egg=hyperpackage\&subdirectory=hyperpackage` in a Jupyter notebook OR
    - `pip install git+https://github.com/gohypergiant/hyperdrive.git@MLSDK472-extract-hp-onnxmodel#egg=hyperpackage\&subdirectory=hyperpackage` from the command line

- check to make sure import hypackage works: 
```
from hyperpackage.utilities import UnpackHyperpack
from hyperpackage.flavors.automl import unpack_hp_univariate_ts as unpack_automl
```

- to test `UnpackHyperpack` class: 
(replace the directory `"./"` with the directory in which an unzipped hyperpack is located
```
unpacker = UnpackHyperpack('./')
unpacker.trial_names # for a list of trial names
# should return a Python list of trial names (will only be one since currently AutoML only saves the best trial) 
unpacker.onnx_filepaths # for a list of file paths (there should also only be one)
```

- to test `UnpackAutoMLHyperpack` class:
```
automl_unpacker = unpack_automl.UnpackAutoMLHyperpack('./', verbose=True)
automlts_onnx_model = automl_unpacker.load_automl_onnx_model()
automl_unpacker.onnx_models
automlts_torch_model = automl_unpacker.convert_onnx_to_automl_torch()
```

_Describe steps to follow to verify the changes that were made._

_If this ticket is a low-fi ticket specify what if anything is testable. A
"not testable" is perfectly acceptable in this case._

_Point out any previously tested functionality that may have been changed
and need another look._

## Notes

_Optional_

_Extra details about the PR that don't fit elsewhere such as dependencies
on another branch, a known issue that needs extra attention, or links._

## Screenshots

image
## Ticket(s)

- [MLSDK-472](https://gohypergiant.atlassian.net/browse/MLSDK-472)

### CODE REVIEW

**Comment by johnhouhyper** [comment]
I was able to test the `UnpackHyperpack` class, and everything looks good.

However, with the `UnpackAutoMLHyperpack` class, I'm getting an import error. I think I see the issue, and I'll leave a comment in the "Files changed" section. For reference, here is the screenshot of the import error:

import_error

I'll continue the review by looking at the rest of the "Files changed".

File: hyperpackage/hyperpackage/flavor/automl/unpack_hp_univariate_ts.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+this script currently only opens up 1 model artifact at a time. 
+Therefore, if multiple model artifacts need to be unpacked, a for-loop
+or iteration over multiple artifact files is required using the code herein.
+"""
+
+from icecream import ic
+import onnx
+from onnx2torch import convert
+
+from utilities import UnpackHyperpack
```
Comment (johnhouhyper) [inline hyperpackage/hyperpackage/flavor/automl/unpack_hp_univariate_ts.py]: When trying to import the `UnpackAutoMLHyperpack` class, I think this is the place where the "import error" is occurring. Might fix it with something like this:

```suggestion
from hyperpackage.utilities import UnpackHyperpack
```

File: hyperpackage/hyperpackage/flavor/automl/unpack_hp_univariate_ts.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+"""This script unpacks hyperpack onnx model artifacts that were 
+originated from Hypergiant's AutoML Univariate TimeSeries.
+NOTE: Since time series is comprised of multiple model artifacts,
+this script currently only opens up 1 model artifact at a time. 
+Therefore, if multiple model artifacts need to be unpacked, a for-loop
+or iteration over multiple artifact files is required using the code herein.
+"""
+
+from icecream import ic
```
Comment (johnhouhyper) [inline hyperpackage/hyperpackage/flavor/automl/unpack_hp_univariate_ts.py:9]: I noticed that you're importing `icecream`, and I see the various `ic` statements. Are we intentionally keeping these in?

File: hyperpackage/hyperpackage/flavor/automl/unpack_hp_univariate_ts.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        ic(type(self.onnx_models))
+
+    def convert_onnx_to_automl_torch(self, ith_model: int = 0):
+        """Convert onnx model to torch model ()
+        :ith_model: if there is greater than one onnx_model in self.onnx_models, use the ith index
+        """
+        automl_model = self.onnx_models[ith_model]
+        self.automl_model = convert(automl_model)
+
+        if self.verbose:
```
Comment (johnhouhyper) [inline hyperpackage/hyperpackage/flavor/automl/unpack_hp_univariate_ts.py:46]: So I see the `verbose` flag being used here in conjunction w/ `icecream`, which makes sense.

File: hyperpackage/hyperpackage/utilities.py [hunk 4]
```diff
@@ -80,19 +88,17 @@ def write_hyperpack_to_s3(
...
+        f"*** COMPLETED: The {s3_object_key} hyperpack was written to the {s3_bucket} S3 bucket ***"
         )
-    )
 
 
 def write_json(dictionary, json_file_path):
     """Writes object to JSON format
     Args:
-        dictionary: python dict to be written to JSON
+        dictionary: pythson dict to be written to JSON
```
Comment (johnhouhyper) [inline hyperpackage/hyperpackage/utilities.py]: Is "pythson" a portmanteau of "python" and "json"?

**Review by johnhouhyper — CHANGES_REQUESTED** [review]
I've left some review comments, please take a look when you get a chance.

File: hyperpackage/hyperpackage/utilities.py (hunk 4 above)
Comment (godsylla) [reply hyperpackage/hyperpackage/utilities.py]: how'd you knOW!? ;)

**Review by johnhouhyper — APPROVED** [review]
LGTM

**Pull request merged**